carsus/io/output/base.py

```diff
@@ -0,0 +1,546 @@
+        levels = levels.rename(columns={'ion_charge': 'ion_number'})
+        levels = levels[['atomic_number', 'ion_number', 'g', 'energy']]
+
+        levels['energy'] = levels['energy'].apply(lambda x: x*u.Unit('cm-1'))
+        levels['energy'] = levels['energy'].apply(
+            lambda x: x.to(u.eV, equivalencies=u.spectral()))
+        levels['energy'] = levels['energy'].apply(lambda x: x.value)
+
+        ground_levels = self.ground_levels
+        ground_levels.rename(
+            columns={'ion_charge': 'ion_number'}, inplace=True)
+
+        # Fixes Ar II duplicated ground level. For Kurucz, ground state
+        # has g=2, for NIST has g=4. We keep Kurucz.
+        mask = (ground_levels['atomic_number'] == 18) & (
+            ground_levels['ion_number'] == 1)
+        ground_levels.loc[mask, 'g'] = 2
+
+        levels = pd.concat([ground_levels, levels], sort=True)
+        levels['level_id'] = range(1, len(levels)+1)
+        levels = levels.set_index('level_id')
+        levels = levels.drop_duplicates(keep='last')
+
+        return levels
+
+    def _get_all_lines_data(self, levels):
+        """ Returns the same output than `AtomData._get_all_lines_data()` """
+        gf = self.gfall_reader
+        df_list = []
+
+        for ion in self.ions:
+
+            try:
+                df = gf.lines.loc[ion]
+
+            except (KeyError, TypeError):
+                continue
+
+            df = df.reset_index()
+            lvl_index2id = levels.set_index(
+                ['atomic_number', 'ion_number']).loc[ion]
+            lvl_index2id = lvl_index2id.reset_index()
+            lvl_index2id = lvl_index2id[['level_id']]
+
+            lower_level_id = []
+            upper_level_id = []
+            for i, row in df.iterrows():
+
+                llid = int(row['level_index_lower'])
+                ulid = int(row['level_index_upper'])
+
+                upper = int(lvl_index2id.loc[ulid])
+                lower = int(lvl_index2id.loc[llid])
+
+                lower_level_id.append(lower)
+                upper_level_id.append(upper)
+
+            df['lower_level_id'] = pd.Series(lower_level_id)
+            df['upper_level_id'] = pd.Series(upper_level_id)
+            df_list.append(df)
+
+        lines = pd.concat(df_list, sort=True)
+        lines['line_id'] = range(1, len(lines)+1)
+        lines['loggf'] = lines['gf'].apply(np.log10)
+
+        lines.set_index('line_id', inplace=True)
+        lines.drop(columns=['energy_upper', 'j_upper', 'energy_lower',
+                            'j_lower', 'level_index_lower',
+                            'level_index_upper'], inplace=True)
+
+        lines.loc[lines['wavelength'] <=
+                  GFALL_AIR_THRESHOLD, 'medium'] = MEDIUM_VACUUM
+        lines.loc[lines['wavelength'] >
+                  GFALL_AIR_THRESHOLD, 'medium'] = MEDIUM_AIR
+        lines['wavelength'] = lines['wavelength'].apply(lambda x: x*u.nm)
```

Comment:
Same as for the energies:
`lines['wavelength'] = (lines['wavelength'].values * u.nm).to(u.AA)`